control-plane/cni/main_test.go

```diff
@@ -0,0 +1,146 @@
+package main
+
+import (
+	"encoding/json"
+	"fmt"
+	"testing"
+
+	"github.com/hashicorp/consul/sdk/iptables"
+	"github.com/stretchr/testify/require"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+func TestWaitForAnnotation(t *testing.T) {
+	t.Parallel()
+	cases := []struct {
+		name       string
+		annotation string
+		pod        func(*corev1.Pod) *corev1.Pod
+		retries    uint64
+		exists     bool
+	}{
+		{
+			name:       "Pod with annotation already existing",
+			annotation: "fooAnnotation",
+			pod: func(pod *corev1.Pod) *corev1.Pod {
+				pod.Annotations["fooAnnotation"] = "foo"
+				return pod
+			},
+			retries: 1,
+			exists:  true,
+		},
+		{
+			name:       "Pod without annotation",
+			annotation: "",
+			pod: func(pod *corev1.Pod) *corev1.Pod {
+				return pod
+			},
+			retries: 1,
+			exists:  false,
+		},
+	}
+	for _, c := range cases {
+		t.Run(c.name, func(t *testing.T) {
+			actual := waitForAnnotation(*c.pod(minimalPod()), c.annotation, c.retries)
+			require.Equal(t, c.exists, actual)
+		})
+	}
+}
+
+func TestParseAnnotation(t *testing.T) {
+	t.Parallel()
+	cases := []struct {
+		name       string
+		annotation string
+		pod        func(*corev1.Pod) *corev1.Pod
```

Comment:
This might be clearer as `configurePod`.

```suggestion
		configurePod        func(*corev1.Pod) *corev1.Pod
```